scalding-core/src/main/java/com/twitter/scalding/cascading_interop/FlowListenerPromise.java

```diff
@@ -34,17 +36,24 @@
     final Promise<T> result = Promise$.MODULE$.<T>apply();
     flow.addListener(new FlowListener() {
       public void onStarting(Flow f) { } // ignore
-      public void onStopping(Flow f) { } // ignore
+      public void onStopping(Flow f) { // in case of runtime exception cascading call onStopping
+        if (!result.isCompleted()) {
```

Comment:
let's use a single atomic call here and do `result.tryFailure`:
https://www.scala-lang.org/api/2.12.3/scala/concurrent/Promise.html#tryFailure(cause:Throwable):Boolean

Reply:
done, thanks